src/transform.py

```diff
@@ -209,33 +334,73 @@ def publishable_speakers(accepted_proposals: set[str]) -> dict[str, PretalxSpeak
     return output
 
 
-def save_publishable_sessions():
+def save_publishable_sessions(publishable: dict[str, PretalxSubmission]):
     path = Config.public_path / "sessions.json"
 
-    publishable = publishable_submissions()
+    for sub in publishable.values():
+        if sub.start is not None:
+            PretalxSubmission.set_talks_in_parallel(sub, publishable)
+            PretalxSubmission.set_talks_after(sub, publishable)
+            PretalxSubmission.set_talks_before(sub, publishable)
 
     data = {k: v.model_dump() for k, v in publishable.items()}
     with open(path, "w") as fd:
         json.dump(data, fd, indent=2)
 
 
-def save_publishable_speakers():
+def save_publishable_speakers(publishable: dict[str, PretalxSubmission]):
     path = Config.public_path / "speakers.json"
 
-    publishable = publishable_submissions()
     speakers = publishable_speakers(publishable.keys())
 
     data = {k: v.model_dump() for k, v in speakers.items()}
     with open(path, "w") as fd:
         json.dump(data, fd, indent=2)
 
 
+def save_all(all_sessions: dict[str, PretalxSubmission]):
+    if not Config.public_path.exists():
+        Config.public_path.mkdir(parents=True)
+
+    save_publishable_sessions(all_sessions)
+    save_publishable_speakers(all_sessions)
+
+
+def check_duplicate_slugs(all_sessions: dict[str, PretalxSubmission]) -> bool:
+    all_speakers = publishable_speakers(all_sessions.keys())
+
+    session_slugs = [s.slug for s in all_sessions.values()]
+    speaker_slugs = [s.slug for s in all_speakers.values()]
+
+    session_duplicates = [
+        slug for slug in set(session_slugs) if session_slugs.count(slug) > 1
+    ]
+    speaker_duplicates = [
+        slug for slug in set(speaker_slugs) if speaker_slugs.count(slug) > 1
+    ]
+
+    if session_duplicates or speaker_duplicates:
+        print("Found duplicate slugs:")
+        for slug in session_duplicates:
+            print(f"Session: {slug}")
+        for slug in speaker_duplicates:
+            print(f"Speaker: {slug}")
+        return False
+    return True
+
+
 if __name__ == "__main__":
+    print(f"Transforming {Config.event} data...")
     print("Checking for duplicate slugs...")
-    assert len(set(s.slug for s in publishable_submissions().values())) == len(
-        publishable_submissions()
-    )
+
+    all_sessions = publishable_submissions()
+
+    if not check_duplicate_slugs(all_sessions) and (
+        len(sys.argv) <= 1 or sys.argv[1] != "--allow-dupes"
+    ):
+        print("Exiting. Use ``make transform ALLOW_DUPES=true`` to continue.")
+        sys.exit(1)
```

Comment:
We should probably handle the submissions with the same names, instead of complaining about it.

Reply:
Now it is handled by appending an incrementing number to the slug: {org_slug}-{1,2,3....}